### PR TITLE
Fix #775 - simple array constraints

### LINKED ISSUE #775: Simple array element enum constraints are broken
String array with enum constraint, but you can still insert any value into the array

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java [hunk 1]
```diff
@@ -445,6 +445,8 @@ private static void copyFields(Fields dest,
                         path.pop();
                     } else {
                         newElement = new SimpleArrayElement(((SimpleArrayElement) sourceEl).getType());
+                        ((SimpleArrayElement)newElement).setConstraints( ((SimpleArrayElement)sourceEl).getConstraints());
```
Comment (dcrissman) [inline metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java]: You could avoid the cast by new'ing SimpleArrayElement as it's own type, then setting it as newElement on the way out.

File: metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java [hunk 2]
```diff
@@ -1489,8 +1489,11 @@ private void convertArrayField(ArrayField field, T fieldObject) {
         putString(items, STR_TYPE, el.getType().getName());
         if (el instanceof ObjectArrayElement) {
             convertObjectArrayElement((ObjectArrayElement) el, items);
+        } else {
+            convertFieldConstraints(items, ((SimpleArrayElement)el).getConstraints());
```
Comment (dcrissman) [inline metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java]: instanceof check?

Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java]: Not necessary, there is only object or simple array element.

File: metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java (hunk 1 above)
Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java]: Done

Comment (dcrissman) [inline metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java]: I am not sure what you changed, but here is an example of what I meant:
```java
SimpleArrayElement simpleElm = new SimpleArrayElement(whatever)
eimpleElm.setConstraints(((SimpleArrayElement)sourceEl).getConstraints())
newElement = simpleElm;
```

Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java]: That's exactly what I did, look at the source

File: metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java (hunk 2 above)
Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java]: github isn't showing the latest here. Look at the latest commit.

File: metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java (hunk 1 above)
Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java]: github is showing the outdated source here. Look at the last commit.

**Comment by dcrissman** [comment]
Sorry, seems I got stuck in a strange time warp. Merging.

**Pull request merged**